modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java

```diff
@@ -170,33 +171,29 @@ protected String getEventName() {
 	}
 
 	protected long[] getGroupIds() {
-		if (_groupIds != null) {
-			return _groupIds;
-		}
-
 		HttpServletRequest httpServletRequest = getRequest();
 
 		ThemeDisplay themeDisplay =
 			(ThemeDisplay)httpServletRequest.getAttribute(
 				WebKeys.THEME_DISPLAY);
 
-		long[] groupIds = null;
-
-		Group group = themeDisplay.getScopeGroup();
+		try {
+			if (ArrayUtil.isEmpty(_groupIds)) {
+				return SiteConnectedGroupGroupProviderUtil.
+					getCurrentAndAncestorSiteAndDepotGroupIds(
+						themeDisplay.getScopeGroupId());
+			}
 
-		if (group.isLayout()) {
-			groupIds = new long[] {group.getParentGroupId()};
-		}
-		else {
-			groupIds = new long[] {group.getGroupId()};
+			return SiteConnectedGroupGroupProviderUtil.
+				getCurrentAndAncestorSiteAndDepotGroupIds(_groupIds);
 		}
-
-		if (group.getParentGroupId() != themeDisplay.getCompanyGroupId()) {
-			groupIds = ArrayUtil.append(
-				groupIds, themeDisplay.getCompanyGroupId());
+		catch (Exception exception) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(exception, exception);
+			}
 		}
 
-		return groupIds;
+		return new long[0];
```

Comment:
Using your logic, what happens if we have something in _groupIds already? What is returned from getGroupIds()?

Reply:
The value we have in _groupIds is used to fetch the needed group ids, if we have something in it, we'll reach this [piece of code](https://github.com/Odrakir1/liferay-portal/blob/6cf606f8e89a91dadc07268c9f6fcb1dd6e2fa92/modules/apps/asset/asset-taglib/src/main/java/com/liferay/asset/taglib/servlet/taglib/AssetTagsSelectorTag.java#L186), which will fetch the ids we need. Just to summarize, _groupIds is used to bring all pertinent ids.